extensions-core/google-extensions/src/main/java/org/apache/druid/storage/google/GoogleStorage.java

```diff
@@ -159,11 +167,23 @@ public void delete(final String bucket, final String path) throws IOException
    * @param bucket GCS bucket
    * @param paths  Iterable for absolute paths of objects to be deleted inside the bucket
    */
-  public void batchDelete(final String bucket, final Iterable<String> paths) throws IOException
+  public void batchDelete(final String bucket, final Iterable<String> paths)
   {
-    List<Boolean> statuses = storage.get().delete(Iterables.transform(paths, input -> BlobId.of(bucket, input)));
+    final List<Boolean> statuses = storage.get().delete(Iterables.transform(paths, input -> BlobId.of(bucket, input)));
```

Comment:
Unlike `deleteFile` call, I don't see where this batch delete call is wrapped in retry logic `GoogleUtils.retryGoogleCloudStorageOperation`, so any transient failures won't be retried? If that's a gap, we should add the retry logic for the batch call too

Reply:
Have now added retry logic at places where batchDelete is used.